src/aihwkit/inference/noise/hermes.py

```diff
@@ -0,0 +1,277 @@
+    aforementioned work.
+
+    Args:
+        prog_coeff: Programming polynomial coeffs in
+            :math:`\sum_i c_i \left(\frac{g_t}{g_\max}\right)^i`
+        g_converter: instantiated class of the conductance converter
+            (defaults to single pair)
+        num_devices: The number of devices that are used to map a weight.
+            Hermes supports either 1 or 2 devices per weight. Defaults to 2.
+        g_max: In :math:`\mu S`, the maximal conductance, ie the value
+            the absolute max of the weights will be mapped to.
+            When `num_devices = 2`, the maximum characterized conductance
+            is 20.7 :math:`\mu S`. When `num_devices = 1` the maximum
+            characterized value is 10.35 :math:`\mu S`. When `None` is passed (by default)
+            the maximum conductance for the corresponding `num_devices` is selected.
+        t_read: Parameter of the 1/f fit (in seconds).
+        t_0: Parameter of the drift fit (first reading time). When `num_devices = 2`
+            that time corresponds to 300s, while in the case that `num_devices = 1`
+            it is 200s. If `None` is passed (by default) the time is selected according
+            to the `num_devices` selection.
+
+            Note:
+                The ``t_inference`` is relative to this time `t0`
+                e.g. t_inference counts from the completion of the programming
+                of a device.
+        prog_noise_scale: Scale for the programming noise.
+        read_noise_scale: Scale for the read and accumulated noise.
+        drift_scale: Scale for the  drift coefficient.
+        prog_coeff_g_max_reference: reference :math:`g_\max` value
+            when fitting the coefficients, since the result of the
+            polynomial fit is given in uS. If
+            ``prog_coeff_g_max_reference`` is not given and
+            `prog_coeffs` are given explicitly, it will be set to
+            ``g_max`` of the conductance converter.
+
+    .. _`Nandakumar et al. ICECS (2019)`: https://ieeexplore.ieee.org/document/8964852
+    .. _`Le Gallo et al. Nat. Electron. (2023)`: https://www.nature.com/articles/s41928-023-01010-1
+    .. _`Khaddam-Aljameh et al. JSSC (2022)`: https://ieeexplore.ieee.org/document/9696185
+    .. _`Vasilopoulos et al. TED (2023)`: https://ieeexplore.ieee.org/document/10281389
+
+    """
+
+    def __init__(
+        self,
+        prog_coeff: Optional[List[float]] = None,
+        g_converter: Optional[BaseConductanceConverter] = None,
+        num_devices: int = 2,
+        g_max: Optional[float] = None,
+        t_read: float = 512.0e-9,
+        t_0: Optional[float] = None,
+        prog_noise_scale: float = 1.0,
+        read_noise_scale: float = 1.0,
+        drift_scale: float = 1.0,
+        prog_coeff_g_max_reference: Optional[float] = None,
+    ):
+        # The only valid options are 1 or 2 devices
+        assert num_devices in [1, 2], "Hermes supports either 1 or 2 devices per weight"
```

Comment:
In theory the number of devices should be set by the `g_converter`.

Reply:
I see here a problem @maljoras Feel like the g_converter is a optional parameter and the num_devices not, what's happened when no exist the g_converter? i think that should be by default 2

Reply:
In the Hermes model, the number of devices has the effects of 
- changing the programming method from normal iterative programming to MSF
- changing the drift model variability based on on-chip measurements.

As such, the g_converter number of devices argument cannot be used for this functionality, leading to a discrete argument. I added some additional details on what the selection of `num_devices` incurs in the docstring of the function (see #691)